mpz-core/src/block.rs

```diff
@@ -121,6 +121,40 @@ impl Block {
         x[0] ^= x[1];
         bytemuck::cast([x[1], x[0]])
     }
+
+    /// Converts a block to a [`GenericArray<u8, U16>`](cipher::generic_array::GenericArray)
+    /// from the [`generic-array`](https://docs.rs/generic-array/latest/generic_array/) crate.
+    #[allow(dead_code)]
+    pub(crate) fn as_generic_array(&self) -> &GenericArray<u8, U16> {
+        (&self.0).into()
+    }
+
+    /// Converts a mutable block to a mutable [`GenericArray<u8, U16>`](cipher::generic_array::GenericArray)
+    /// from the [`generic-array`](https://docs.rs/generic-array/latest/generic_array/) crate.
+    pub(crate) fn as_generic_array_mut(&mut self) -> &mut GenericArray<u8, U16> {
+        (&mut self.0).into()
+    }
+
+    /// Converts a slice of blocks to a slice of [`GenericArray<u8, U16>`](cipher::generic_array::GenericArray)
+    /// from the [`generic-array`](https://docs.rs/generic-array/latest/generic_array/) crate.
+    #[allow(dead_code)]
+    pub(crate) fn as_generic_array_slice(slice: &[Self]) -> &[GenericArray<u8, U16>] {
+        // # Safety
+        // This is always safe because `Block` and `GenericArray<u8, U16>` have the same memory layout.
+        // See https://github.com/fizyk20/generic-array/blob/37dc6aefc3ed5c423ad7402d4febf06a3e78a223/src/lib.rs#L838-L845
+        // TODO: Use methods provided by `generic-array` once 1.0 is released.
+        unsafe { std::mem::transmute(slice) }
```

Comment:
We rely here on an implementation detail of an external library. What happens if the memory layout of `GenericArray` is changed upstream?